org.jacoco.agent.rt.test/src/org/jacoco/agent/rt/internal/output/FileOutputTest.java

```diff
@@ -72,4 +78,48 @@ public void testInvalidDestFile() throws Exception {
 		controller.startup(options, new RuntimeData());
 	}
 
+	@Test(expected = OverlappingFileLockException.class)
```

Comment:
We can deal with already existing usages of `@Test(expected = ...)` separately, but for new ones I prefer try/catch idiom - see https://github.com/junit-team/junit4/wiki/Exception-testing#trycatch-idiom

Reply:
Wasn't the try/catch idiom for JUnit 3.x? Why do you prefer this? I would rather move to the JUnit 5 idiom `assertThrows(extype, sut)` once we have a Java 8 baseline.

Reply:
@marchof For the reasons explained on page https://github.com/junit-team/junit4/wiki/Exception-testing#trycatch-idiom Among which: With `@Test(expected = ...)` and `ExpectedException` easy to overlook which code in test is executed and which is not due to exception, while try/catch allows to more precisely specify at which place exception is expected. When switching to `assertThrows` will you prefer to wrap entire method into `assertThrows` or just concrete part? :wink: so use of try/catch now will also kinda help to migrate on `assertThrows` one day. try/catch also allows to test message of exception if needed, and we already have places where we need and do use try/catch, thus also for consistency.